File: backend/app/api/graph.py
"""知识图谱 API 路由（Phase 3）。

只读返回知识库内抽取的三元组图快照，供前端可视化，
委托 `graph_store_service.graph_snapshot` 聚合节点与边。
"""

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from ..core.database import get_db
from ..services.graph_store_service import graph_snapshot

router = APIRouter(prefix="/api/knowledge-bases", tags=["graph"])


@router.get("/{kb_id}/graph")
async def get_knowledge_graph(kb_id: str, limit: int = 80, db: AsyncSession = Depends(get_db)):
    """获取知识库轻量图谱快照。

    参数:
        kb_id: 知识库 ID。
        limit: 最大节点数，内部 capped 为 200。
        db: 数据库会话。

    返回:
        含 nodes/edges/relation_count 及 empty 标记的字典。
    """
    snap = await graph_snapshot(db, kb_id, limit_nodes=min(limit, 200))
    if snap["relation_count"] == 0:
        return {**snap, "empty": True}
    return {**snap, "empty": False}

File: frontend/src/api/graph.ts
/**
 * 知识图谱 API
 * 获取实体-关系快照供力导向图展示
 * 主要导出：GraphNode、GraphEdge、KnowledgeGraphSnapshot、graphApi
 */
import request from './request'

/** 图谱节点（实体） */
export interface GraphNode {
  id: string
  name: string
}

/** 图谱边（关系三元组） */
export interface GraphEdge {
  source: string
  target: string
  predicate: string
  chunk_id?: string
}

/** 知识库图谱快照 */
export interface KnowledgeGraphSnapshot {
  nodes: GraphNode[]
  edges: GraphEdge[]
  relation_count: number
  node_count: number
  empty?: boolean
}

export const graphApi = {
  /** 获取知识库实体关系子图，limit 限制边数 */
  getSnapshot: (kbId: string, limit = 80) =>
    request.get<KnowledgeGraphSnapshot>(`/api/knowledge-bases/${kbId}/graph`, {
      params: { limit },
    }),
}
